dnserver/main.py

```diff
@@ -146,18 +140,26 @@ class DNSServer:
     def __init__(
         self, zones_file: str | Path, *, port: int | str | None = DEFAULT_PORT, upstream: str | None = DEFAULT_UPSTREAM
     ):
-        self.zones_file = zones_file
         self.port: int = DEFAULT_PORT if port is None else int(port)
         self.upstream: str | None = upstream
         self.udp_server: LibDNSServer | None = None
         self.tcp_server: LibDNSServer | None = None
 
+        self.records: Records = load_records(zones_file)
```

Comment:
`__init__` should take `records: Records` as an argument.

Then add a class method like:

```py
@classmethod
def from_toml(zones_file: str | Path, *, port: int | str | None = DEFAULT_PORT, upstream: str | None = DEFAULT_UPSTREAM) -> 'DNSServer':
    records = load_records(zones_file)
    return DNSServer(records, port=port, upstream=upstream)
```

Reply:
Done. I made `records` an optional argument in case one wants to only use `add_record` to set zones.

Reply:
It's not currently an optional argument, you need to give it a default value.

Reply:
Ok this is fixed.